# **0. COLLECTING DATA**

## Import thư viện cần thiết

In [1]:
import pandas as pd

## Đọc dữ liệu

In [4]:
df = pd.read_csv("../data/youtube_vn_music.csv")
print("Tổng quan dữ liệu:")
df.info()

Tổng quan dữ liệu:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10615 entries, 0 to 10614
Data columns (total 23 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   video_id                  10615 non-null  object 
 1   published_at              10615 non-null  object 
 2   channel_id                10615 non-null  object 
 3   channel_title             10615 non-null  object 
 4   title                     10615 non-null  object 
 5   description               9536 non-null   object 
 6   tags                      8725 non-null   object 
 7   category_id               10615 non-null  int64  
 8   view_count                10615 non-null  int64  
 9   like_count                10018 non-null  float64
 10  comment_count             10451 non-null  float64
 11  duration                  10615 non-null  object 
 12  definition                10615 non-null  object 
 13  caption                   10615 non-null  

In [3]:
df.head(5)

,video_id,published_at,channel_id,channel_title,title,description,tags,category_id,view_count,like_count,...,caption,licensed_content,privacy_status,made_for_kids,tags_count,topic_categories,search_query,channel_subscriber_count,channel_view_count,channel_video_count
0,sZrIbpwjTwk,2025-08-17T05:00:48Z,UC2fu6CiFfNYz5UFORvFyc0w,Vie Channel - MUSIC,ẾCH NGOÀI ĐÁY GIẾNG - Phương Mỹ Chi hoá thân t...,ẾCH NGOÀI ĐÁY GIẾNG - Phương Mỹ Chi hoá thân t...,htv2|vie channel|music vie channel|em xinh say...,10,82407446,234808.0,...,False,True,public,False,22,https://en.wikipedia.org/wiki/Music|https://en...,bài,1710000.0,2.303825e+09,2519.0
1,Qgzb0musH3A,2022-08-24T12:00:12Z,UCIKpxDD9CUS7vBCJte7l-UQ,Như Hoa Official,Bài hát đang Hót🔥TRỞ VỀ_Như Hoa|Bài Hát Lấy Đi...,Bài Hát Đang Hót 🔥 TRỞ VỀ | Như Hoa ( Thần Tượ...,Bài Hát hay nhất về Mẹ|Bài hát ý nghĩa nhất về...,10,6188111,6227.0,...,False,True,public,False,15,https://en.wikipedia.org/wiki/Music|https://en...,bài,246000.0,1.817580e+08,842.0
2,hmc5CFN8bmk,2024-01-11T11:00:08Z,UCUjRPenj3PuEJ2X0Kju9Tug,Lâm Hùng Official,Duyên Hồng - Lâm Hùng || Bài hát hay nhất mùa ...,--------------------\nKênh Youtube chính thức ...,lam hung|lâm hùng|lâm hùng official|lam hung o...,10,16959972,NaN,...,False,False,public,False,18,https://en.wikipedia.org/wiki/Music|https://en...,bài,509000.0,2.921241e+08,494.0
3,FxX5pTGW2dQ,2023-11-18T09:30:05Z,UCbwr_JmPZbfYsHvtWR0j6bw,Nhạc Thiếu Nhi,Bài Hát ABC - Học Bảng Chữ Cái ABC - Nhạc Thiế...,Bài Hát ABC - Học Bảng Chữ Cái ABC - Nhạc Thiế...,nhạc thiếu nhi|nhạc thiếi|u nhi vui nhộn|nhạc ...,10,102619561,88130.0,...,False,True,public,True,23,https://en.wikipedia.org/wiki/Music,bài,3190000.0,1.569856e+09,519.0
4,xrfd0C7aQf0,2025-03-15T06:46:25Z,UCWEYgC77_ZlbDxStQyzOwfA,Hiếu Organ,Bài Ca Kỷ Niệm Karaoke Tone Nữ ( Bm ) Âm Thanh...,Karaoke Bài Ca Kỷ Niệm Karaoke Tone Nữ ( Bm ) ...,Karaoke|KARAOKE|karaoke nhạc sống|karaoke nhac...,10,631762,NaN,...,False,True,public,False,21,https://en.wikipedia.org/wiki/Music|https://en...,bài,4750000.0,5.909387e+09,4982.0


## Tổng quan đặc trưng (features)

Bộ dữ liệu chứa 10615 mẫu video với 23 đặc trưng:

- `video_id` - Mã định danh duy nhất của video trên YouTube.
- `title` - Tiêu đề đầy đủ của video.
- `description` - Nội dung mô tả chi tiết bên dưới video.
- `category_id` - Mã số danh mục của video.
- `view_count` - Tổng số lượt xem của video.
- `like_count` - Tổng số lượt thích của video.
- `comment_count` - Tổng số lượt bình luận của video.
- `duration` - Thời lượng video (theo chuẩn ISO 8601).
- `published_at` - Thời điểm video được đăng lên.
- `definition` - Chất lượng phát video (cao nhất).
- `caption` - Trạng thái video có phụ đề hay không (có/không).
- `licensed_content` - Tình trạng bản quyền của video (có/không).
- `privacy_status` - Chế độ riêng tư của video (private/unlisted/public).
- `made_for_kids` - Nội dung cho trẻ em (có/không).
- `tags` - Danh sách các từ khóa/thẻ gắn kèm video.
- `tags_count` - Số lượng từ khóa/thẻ gắn kèm video.
- `topic_categories` - Các chủ đề liên quan (dạng link Wikipedia).
- `search_query` - Truy vấn dùng để tìm video.
- `channel_id` - Mã định danh duy nhất của kênh đăng tải video.
- `channel_title` - Tên kênh đăng tải video.
- `channel_subscriber_count` - Số lượt đăng ký của kênh đăng tải video.
- `channel_view_count` - Tổng số lượt xem của kênh đăng tải video.
- `channel_video_count` - Tổng số video của kênh đăng tải video.